app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/NotesApi.kt

```diff
@@ -0,0 +1,120 @@
+     *
+     * @return the updated reopened note
+     */
+    fun reopen(id: Long, reason: String? = null): Note = notesDao.reopen(id, reason).toNote()
+
+    /**
+     * Close aka resolve the note with the given id and reason.
+     *
+     * @param id id of the note
+     * @param reason comment to be added to the note as a reason for it being closed. Optional.
+     *
+     * @throws OsmConflictException if the note has already been closed.
+     * @throws OsmAuthorizationException if this application is not authorized to write notes
+     *                                   (Permission.WRITE_NOTES)
+     * @throws OsmNotFoundException if the note with the given id does not exist (anymore)
+     *
+     * @return the closed note
+     */
+    fun close(id: Long, reason: String? = null): Note = notesDao.close(id, reason).toNote()
+
+    /**
+     * @param id id of the note
+     *
+     * @return the note with the given id. null if the note with that id does not exist (anymore).
+     */
+    operator fun get(id: Long): Note? = notesDao.get(id)?.toNote()
+
+    /**
+     * Retrieve all notes in the given area and feed them to the given handler.
+     *
+     * @see .getAll
+     */
+    fun getAll(bounds: BoundingBox, handler: Handler<Note>, limit: Int, hideClosedNoteAfter: Int) {
+        getAll(bounds, null, handler, limit, hideClosedNoteAfter)
+    }
+
+    /**
+     * Retrieve those notes in the given area that match the given search string
+     *
+     * @param bounds the area within the notes should be queried. This is usually limited at 25
+     *               square degrees. Check the server capabilities.
+     * @param search what to search for. Null to return everything.
+     * @param handler The handler which is fed the incoming notes
+     * @param limit number of entries returned at maximum. Any value between 1 and 10000
+     * @param hideClosedNoteAfter number of days until a closed note should not be shown anymore.
+     *                            -1 means that all notes should be returned, 0 that only open notes
+     *                            are returned.
+     *
+     * @throws OsmQueryTooBigException if the bounds area is too large
+     * @throws IllegalArgumentException if the bounds cross the 180th meridian
+     */
+    fun getAll(bounds: BoundingBox, search: String?, handler: Handler<Note>, limit: Int, hideClosedNoteAfter: Int) =
+        notesDao.getAll(bounds.toOsmApiBoundingBox(), search, handler.toOsmApiNoteHandler(), limit, hideClosedNoteAfter)
+
+    /**
+     * Get a number of notes that match the given filters.
+     *
+     * @param handler The handler which is fed the incoming notes
+     * @param filters what to search for. I.e. `QueryNotesFilters().byUser(123).limit(1000)`
+     */
+    fun find(handler: Handler<Note>, filters: QueryNotesFilters?) =
+        notesDao.find(handler.toOsmApiNoteHandler(), filters)
```

Comment:
Is this even used in the code? The wrapper only needs to offer what is actually used in the code.